src/components/home/elements/SplitToken.vue

```diff
@@ -0,0 +1,671 @@
+        },
+        coldAddresses: {
+            type: Object,
+            default: function() {},
+            require: true// Can not be removed ,cause this is an inherited property may be used somewhere
+        },
+        maxSupply: {
+            type: BigNumber,
+            default: function() {
+            },
+            require: true
+        },
+        addresses: {
+            type: Object,
+            default: function() {},
+            require: true
+        },
+        walletType: {
+            type: String,
+            default: 'hotWallet',
+            require: true
+        },
+        tokenId: {
+            type: String,
+            default: '',
+            require: true
+        },
+        maker: {
+            type: String,
+            default: '',
+            require: true
+        },
+        address: {
+            type: String,
+            default: ''
+        }
+    },
+    computed: {
+        contractId() {
+            return transaction.tokenIDToContractID(this.tokenId)
+        },
+        defaultAddress() {
+            return Vue.ls.get('address')
+        },
+        defaultColdAddress() {
+            if (this.noColdAddress) return ''
+            return Object.keys(this.coldAddresses)[0]
+        },
+        userInfo() {
+            return JSON.parse(window.localStorage.getItem(this.defaultAddress))
+        },
+        secretInfo() {
+            return JSON.parse(
+                seedLib.decryptSeedPhrase(this.userInfo.info, Vue.ls.get('pwd')))
+        },
+        seedPhrase() {
+            return seedLib.decryptSeedPhrase(this.secretInfo.encrSeed, Vue.ls.get('pwd'))
+        },
+        wordList() {
+            return this.seedPhrase.split(' ')
+        },
+        noColdAddress() {
+            return Object.keys(this.coldAddresses).length === 0 && this.coldAddresses.constructor === Object
+        },
+        dataObject() {
+            return {
+                protocol: PROTOCOL,
+                api: this.coldApi(),
+                opc: OPC_FUNCTION,
+                address: this.address,
+                senderPublicKey: this.coldAddresses[this.address].publicKey,
+                fee: this.fee * VSYS_PRECISION,
+                feeScale: FEE_SCALE,
+                timestamp: Date.now(),
+                attachment: '',
+                contractId: this.contractId,
+                functionId: SPLIT_FUNCIDX,
+                function: transaction.prepareSplit(BigNumber(this.newUnity)),
+                functionExplain: 'Set token unity to ' + this.newUnity
+            }
+        }
+    },
+    methods: {
+        coldApi: function() {
+            return API_VERSION
+        },
+        isValidMaker: function(addr) {
+            return addr === this.maker
+        },
+        isSubmitDisabled() {
+            return !(this.isValidMaker(this.address) && this.isValidUnity() && !this.isInsufficient() && this.isSplit)
+        },
+        isBiggerThanMax() {
```

Comment:
maxSupply in chain is store as none unit case

what is this function check for?

Reply:
this function is used to check whether value is greater than 2^63-1

Reply:
already removed